app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx

```diff
@@ -15,158 +15,57 @@
-        return {
-          ...connectorType,
-          SVG: Spanner,
-        };
-      } else if (connectorType.name === 'Kafka') {
-        return {
-          ...connectorType,
-          SVG: Kafka,
-        };
-      } else if (connectorType.name === 'SQL Server') {
-        return {
-          ...connectorType,
-          SVG: SQLServer,
-        };
-      } else if (connectorType.name === 'MySQL') {
-        return {
-          ...connectorType,
-          SVG: MySQL,
-        };
-      } else if (connectorType.name === 'Oracle') {
-        return {
-          ...connectorType,
-          SVG: Oracle,
-        };
-      } else if (connectorType.name === 'PostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: PostgreSQL,
-        };
-      } else if (connectorType.name === 'File') {
-        return {
-          ...connectorType,
-          SVG: ImportDatasetIcon,
-        };
-      } else if (connectorType.name === 'CloudSQLMySQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLMySQL,
-        };
-      } else if (connectorType.name === 'CloudSQLPostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLPostGreSQL,
-        };
-      } else {
-        return {
-          ...connectorType,
-          SVG: BigQuery,
-        };
-      }
-    });
-
-    /* remove the other connector Types based on getCategorized connections */
-
-    fetchedConnectorTypesFromAPI = fetchedConnectorTypesFromAPI.filter((obj) =>
-      connectorTypeWithConnections.find((each) => each === obj.name)
-    );
-
-    fetchedConnectorTypesFromAPI.unshift({
-      name: 'Imported Datasets',
-      type: 'default',
-      category: 'default',
-      description: 'All Connections from the List',
-      artifact: {
-        name: 'allConnections',
-        version: 'local',
-        scope: 'local',
-      },
-
-      SVG: ImportDatasetIcon,
-    });
+  const classes = useStyles();
+  const connectorTypes: IConnectorArray[] = connectorsData.connectorTypes;
 
-    setConnectorTypes({
-      fetchedConnectorTypes: fetchedConnectorTypesFromAPI,
-    });
+  const updateState = (updatedState) => {
+    setConnectorsData(updatedState);
   };
 
   useEffect(() => {
-    getConnectorTypesNames();
+    getWidgetData(updateState);
   }, []);
 
-  const classes = useStyles();
-  const fetchedConnectorTypes = connectorTypes.fetchedConnectorTypes;
+  let startIndex = 0;
+  let endIndex = 2;
+  // Here we are finding out the connections are exist in the app or not
+  if (connectorTypes.length > 2) {
+    startIndex = 1; // This line is writtern to eliminate the add-connection cards's data from the array
+    endIndex = 5;
+  }
+
   return (
     <Box className={classes.wrapper} data-testid="wrangle-card-parent">
-      {fetchedConnectorTypes.map((item, index) => {
+      {/* Here we are only showing top 4 connectors on home page */}
+      {connectorTypes.slice(startIndex, endIndex).map((item, index) => {
```

Comment:
null check

Reply:
Connector types array always have a length of 2. So, null check isn't required here.